bot/cogs/pin_cog.py

```diff
@@ -0,0 +1,69 @@
+import discord
+from discord.ext import commands
+
+import bot.extensions as ext
+from bot.consts import Colors
+from bot.data.class_repository import ClassRepository
+from bot.data.pin_repository import PinRepository
+from bot.messaging.events import Events
+from bot.sock_bot import SockBot
+from bot.utils.helpers import deletable_error_embed
+
+PIN_COMMAND_COOLDOWN = 120  # 2 minutes
+
+
+class PinCog(commands.Cog):
+
+    def __init__(self, bot: SockBot) -> None:
+        self.bot = bot
+        self.class_repo = ClassRepository()
+        self.pin_repo = PinRepository()
+
+    @ext.command(case_insensitive=True)
+    # @commands.cooldown(1, PIN_COMMAND_COOLDOWN, commands.BucketType.user)
+    @ext.short_help('Open a pin request.')
+    @ext.long_help('Open a request to get a message pinned in your class channel.')
+    @ext.example(['pin https://discord.com/...'])
+    async def pin(self, ctx, message: discord.Message | None = None) -> None:
+        if not message and ctx.message.reference:
+            message = await ctx.channel.fetch_message(ctx.message.reference.message_id)
+        if not message:
+            await deletable_error_embed(self.bot, ctx, 'Reply to a message or give the ID or link to a message.')
+            return
+        if message.pinned:
+            await deletable_error_embed(self.bot, ctx, 'The message given is already pinned.')
+            return
+        if message.author.bot:
+            await deletable_error_embed(self.bot, ctx, 'Cannot pin a bot message.')
```

Comment:
Why not

Reply:
I do not see a situation in a class channel that a student would have to pin a message from a bot, but if you can think of one I will reconsider it. :)